### ISSUE TITLE
wsgi environ access

### ISSUE THREAD

**Issue by ruthenium** [toastdriven/itty#12]
To use itty-based applications in a wsgi pipeline with another, itty app should have more access to environ dict.
The most **important** is a 'SCRIPT_NAME' param, because it is used to construct application urls.
For example, I've implemented it in a ruthenium@23ceba7c445b33fdd110, where I've just added the script_name and user params to request:

``` python

self.script_name = self._environ.get('SCRIPT_NAME', '')
self.user = self._environ.get('REMOTE_USER', None)

```

But maybe the request should have access to the private environ variable using something like 'getenv' or etc?..

**Comment by toastdriven** [comment]
I guess I'm confused on this. Couldn't you get the same information elsewhere by doing `request._environ.get('SCRIPT_NAME', '')`? If you could explain the use case a little better (like in conjunction with what other software does this fall down), that'd help me a lot.

**Comment by ruthenium** [comment]
Of course I can do it, but the issue is that the class members starting with _ are supposed to be private in python. Yes, they are accessible from the outside, but I thought it is not a good practice to do so. Anyway, I've created this issue mainly in order to find a better way to access the environ — should it be accessed directly, as you described, or through some helpers, that, as I thought would be more ideologically correct.

**Comment by toastdriven** [comment]
Fair enough. That variable wasn't going to change anytime soon but I can understand not feeling comfortable using it. Let me think on it a bit & I'll work up a better way to access things.

**Comment by ruthenium** [comment]
OK, thanks for your effort!

**Comment by toastdriven** [comment]
Fixed in SHA: c3b716f8ba764a592e13c43ce0e065a0ff8b830f & released as v0.8.0.
